sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -134,37 +144,168 @@ export class IER12ProcessingService {
+        hasProvincialDefaultRestriction,
+        hasProvincialOveraward: studentOverawardsBalance
+          ? studentOverawardsBalance[FullTimeAwardTypes.BCSL] > 0
+          : false,
+        hasFederalOveraward: studentOverawardsBalance
+          ? studentOverawardsBalance[FullTimeAwardTypes.CSLF] > 0
+          : false,
+        hasRestriction,
+        assessmentTriggerType: pendingAssessment.triggerType,
+        scholasticStandingChangeType: scholasticStanding?.changeType,
+        assessmentDate: pendingAssessment.assessmentDate,
+        hasPartner,
+        parentalAssets:
+          pendingAssessment.workflowData.calculatedData.parentalAssets,
+        coeStatus: disbursement.coeStatus,
+        disbursementScheduleStatus: disbursement.disbursementScheduleStatus,
+        earliestDateOfDisbursement: new Date(disbursement.disbursementDate),
+        dateOfDisbursement: disbursementReceipt?.disburseDate
+          ? new Date(disbursementReceipt.disburseDate)
+          : null,
+        disbursementCancelDate:
+          disbursement.disbursementScheduleStatus ===
+          DisbursementScheduleStatus.Cancelled
+            ? disbursement.updatedAt
+            : null,
+        disbursementAwards: this.getDisbursementAwards(
+          disbursement.disbursementValues,
+        ),
+      };
+      ier12Records.push(ier12Record);
+    }
+    return ier12Records;
+  }
+
+  /**
+   * Get all disbursement award details for IER.
+   * @param disbursementValues disbursement values.
+   * @returns award details.
+   */
+  private getDisbursementAwards(
+    disbursementValues: DisbursementValue[],
+  ): IERAward[] {
+    return disbursementValues.map<IERAward>((disbursementValue) => ({
+      valueType: disbursementValue.valueType,
+      valueCode: disbursementValue.valueCode,
+      valueAmount: disbursementValue.valueAmount,
+    }));
+  }
+
+  /**
+   * Get all disbursement award details for IER of all disbursements that belong
+   * to a given assessment.
+   * @param disbursementSchedules disbursement schedules.
+   * @returns assessment award details.
+   */
+  private getAssessmentAwards(
+    disbursementSchedules: DisbursementSchedule[],
+  ): IERAward[] {
+    const assessmentAwards: IERAward[] = [];
+    for (const disbursement of disbursementSchedules) {
+      const disbursementAwards = this.getDisbursementAwards(
+        disbursement.disbursementValues,
+      );
+      assessmentAwards.push(...disbursementAwards);
+    }
+    return assessmentAwards;
+  }
+
+  /**
+   * Check if student has an active restriction.
+   * @param studentRestrictions student restrictions
+   * @param options check restriction options:
+   * - `restrictionCode`: restriction code.
+   * @returns value which indicates
+   * if a student has active restriction.
+   */
+  private checkActiveRestriction(
+    studentRestrictions: StudentRestriction[],
+    options?: { restrictionCode?: string },
+  ): boolean {
+    return studentRestrictions?.some(
```

Comment:
correct me if I am wrong, if the `restrictionCode` is not passed in the option, then the responsibility of this method is to return `true` if there is any active restriction right? if so, `(true && (false || false))` will be `false`. let me know, if I am missing something

Reply:
Particularly on that scenario where a student has active restriction and the option is not passed it will be 

true && (true || not evaluated) = true 

![image](https://github.com/bcgov/SIMS/assets/54600590/f1c34d4d-f7cc-434d-8cf1-f4981e22806a)

Reply:
yup.. You are correct, I overlooked  the `!` in `!options?.restrictionCode`. Thanks